game-core/src/main/java/games/strategy/triplea/ui/screen/TileManager.java

```diff
@@ -371,6 +376,23 @@ private void drawUnits(final Territory territory, final MapData mapData, final S
     }
   }
 
+  private static List<UnitCategory> getSortedUnitCategories(final Territory t, final GameData data) {
+    final List<UnitCategory> categories = new ArrayList<>(UnitSeperator.categorize(t.getUnits().getUnits()));
+    final List<UnitType> xmlUnitTypes = new ArrayList<>(data.getUnitTypeList().getAllUnitTypes());
+    categories.sort(Comparator
+        .comparing(UnitCategory::getOwner, Comparator
+            .comparing((final PlayerID p) -> !p.equals(t.getOwner()))
+            .thenComparing(p -> !Matches.isAllied(p, data).test(t.getOwner()))
+            .thenComparing(data.getPlayerList().getPlayers()::indexOf))
+        .thenComparing(Comparator.comparing(uc -> Matches.unitTypeCanMove(uc.getOwner()).test(uc.getType())))
+        .thenComparing(UnitCategory::getType, Comparator
+            .comparing((final UnitType ut) -> !Matches.unitTypeCanNotMoveDuringCombatMove().test(ut))
+            .thenComparing(ut -> !Matches.unitTypeIsLand().test(ut))
+            .thenComparing(ut -> !UnitAttachment.get(ut).getIsSea())
+            .thenComparing(xmlUnitTypes::indexOf)));
+    return categories;
+  }
```

Comment:
Now that this method is really just sorting a "given list" (using mocks etc.) it'd be really good if there was a Unit Test for this little detail at least.

Reply:
Added unit test.